catroid/src/main/java/org/catrobat/catroid/web/CatrobatWebClient.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2019 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.web
+
+import android.support.annotation.VisibleForTesting
+import okhttp3.ConnectionSpec
+import okhttp3.FormBody
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import java.io.IOException
+
+object CatrobatWebClient {
+    val client: OkHttpClient = OkHttpClient.Builder()
+        .connectionSpecs(listOf(ConnectionSpec.MODERN_TLS))
+        .build()
+}
+
+class WebCallBuilder {
+    companion object {
+        const val CONNECTION_RETRIES = 5
+    }
+
+    val request: Request
+
+    @VisibleForTesting
+    constructor(request: Request) {
+        this.request = request
+    }
+
+    constructor(url: String, postValues: Map<String, String>) {
+        val formEncodingBuilder = FormBody.Builder()
+        for ((key, value) in postValues) {
+            formEncodingBuilder.add(key, value)
+        }
+        request = Request.Builder()
+            .url(url)
+            .post(formEncodingBuilder.build())
+            .build()
+    }
+
+    @Throws(WebconnectionException::class)
+    fun performCall(okHttpClient: OkHttpClient): String {
```

Comment:
move to extension fun of OkHttpClient. call like `client.performCallWith(request: Request)`